src/parse.ts

```diff
@@ -358,7 +360,19 @@ export default class Parser {
 					continue;
 				}
 
-				value += escapeable[char] || char;
+				if (char === 'x' || char === 'u') {
+					const start = this.index;
+					const end = this.index += (char === 'x' ? 2 : 4);
+
+					const code = this.str.slice(start, end);
+					if (!hex.test(code)) this.error(`Invalid ${char === 'x' ? 'hexadecimal' : 'Unicode'} escape sequence`, start);
```

Comment:
I think this will only check whether _none_ of the characters are valid, not whether they're all valid.

Reply:
d'oh. fixed